### DISCUSSION TITLE
Unexplainable Damping

### DISCUSSION THREAD

**Discussion by iBims1JFK** [stack-of-tasks/pinocchio#1763] [Q&A]
Hey, I am really sorry to bother you again, but I encountered another problem which I cannot solve after endless hours of investigation. I just hope, that this is a problem that you may have seen so far and thus can help me. As described in #1696, I am trying to stabilize a spherical pendulum on a 4-DOF Barrett WAM. Futhermore, I am using PyBullet for simulating the system and Pinocchio for transforming my task space controller into a joint space controller. 

The controller I have modeled basically consists out of two controllers, which try to control two cart poles which are perpendicular. I tested these in a numerical simulation, which showed that this approach works. It also works in PyBullet when simulating just a spherical cart pendulum in the task space. 

Moreover, I have tried to build the joint space controller by using this formula $\tau=MJ^{+}(\ddot{x_r}-\dot{J}\dot{q})+C+g$. 
This far it works really good, but for some reason only for only one direction.

Pendulum falling towards phi
![phi_30](https://user-images.githubusercontent.com/35470554/194559558-1142c4a1-eb22-4c79-bfba-74374a59d00f.png)
Pendulum falling towards negative phi
![phi_-30](https://user-images.githubusercontent.com/35470554/194560366-bed2a26b-7cf2-4434-bcff-7fcaacb60942.png)

Pendulum falling towards theta
![theta_30](https://user-images.githubusercontent.com/35470554/194559572-59974e7b-1ed4-44c2-a803-c52e0ac8bf2b.png)
Pendulum falling towards negative theta
![theta_-30](https://user-images.githubusercontent.com/35470554/194559593-b7b49a70-d730-402c-ad75-0be3d21604f1.png)

As you can see, the pendulum is stable when it is falling along phi. On the other hand when the pendulum is falling along theta, the endeffector is not returning to its home position and you can see (especially in the positive theta plot) that the pendulum is not straight upwards. To me it seems something is damping the system but only on one side because it does not matter in which direction the pendulum was initially aligned, it always ends up on the right side.

It is a very strange behavior, maybe some of you have an idea whats the problem. Thank you!

**Comment by jcarpent** [comment]
@iBims1JFK It seems that your issue is not directly related to Pinocchio, but to a broader topic related to your activities.
I would like to move it to a discussion.
In addition, I would encourage you sharing your code with other people.
From our side, we received hundreds of solicitations like yours every year, which drastically limits our capacity to answer your current issue while having to develop and maintain many open-source software.
I hope you will understand our position.
